numpy/lib/function_base.py

```diff
@@ -660,7 +660,7 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
     if mn > mx:
         raise ValueError(
             'max must be larger than min in range parameter.')
-    if not np.all(np.isfinite([mn, mx])):
+    if not np.all(np.isfinite(np.hstack([mn, mx]))):
```

Comment:
This can of worms would be better solved with `np.isfinite(mn) and np.isinfinite(mx)`